src/main/java/org/jenkinsci/maven/plugins/hpi/TestDependencyMojo.java

```diff
@@ -550,6 +549,12 @@ private static Map<String, String> scanWar(File war, MavenSession session, Maven
                         if (version == null) {
                             throw new IllegalArgumentException("Failed to determine version for " + name);
                         }
+                        // handle any extra info in snapshots e.g. " (private-abcd1234-username)"
```

Comment:
Whilst we could use `Implementation-Version` and then fall back - some plugins (`gradle`) do not provide this due to tooling or being too old (this was added approx. 2 years ago).  Given the need to fallback, it is easier just to handle the legacy case here.